external-crates/move/crates/move-compiler/src/shared/mod.rs

```diff
@@ -365,6 +365,16 @@ impl CompilationEnv {
     }
 
     pub fn add_diag(&mut self, mut diag: Diagnostic) {
+        if diag.info().severity() <= Severity::NonblockingError
+            && self
+                .diags
+                .any_syntax_error_with_primary_loc(diag.primary_loc())
```

Comment:
I did not think that it would matter perf-wise (e.g., compared to compilation time) and it's tough to do it consistently at reporting time as diags can be obtained in a variety different ways. For example, the symbolicator may get them from results of compiler `run` functions (at different AST levels) which is different from what happens in CLI.

Reply:
As per separate discussion, added a `TODO` there for now